### PR TITLE
renamed directory, splitting contract VERSION

### PR DESCRIPTION
**Pull request by Maxstgt**
renamed directory forms to form. 
splitting up contract form into contract and contract VERSION. 
and refactoring of party-name component
ref #1801

### CODE REVIEW

File: libs/contract/src/lib/contract/form/party-name/party-name.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+            <mat-form-field appearance="outline">
+                <mat-label>{{ type | titlecase }} Name</mat-label>
+                <input matInput type="text" formControlName="displayName" autocomplete="off">
+            </mat-form-field>
+        </form>
+        <ng-container *ngIf="!isLast; else add">
+            <button (click)="form.removeAt(i)" mat-mini-fab color="warn">
+                <mat-icon>delete</mat-icon>
+            </button>
+            <span>Remove {{ type | titlecase  }}</span>
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/contract/form/party-name/party-name.component.html]: Put this span in a tooltip when it's a row.

File: libs/contract/src/lib/contract/form/party-name/party-name.component.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+@Component({
+  selector: '[form] contract-form-party-name',
+  templateUrl: 'party-name.component.html',
+  styleUrls: ['party-name.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ContractFormPartyNameComponent {
+  @Input() form: FormList<any, PartyDetailsForm>;
+  @Input() type?: 'licensee' | 'licensor' = 'licensee';
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/contract/form/party-name/party-name.component.ts]: remove "?" as you are setting the "license" by default

File: libs/contract/src/lib/contract/form/party-name/party-name.component.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  public addEntity(){
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/contract/form/party-name/party-name.component.ts]: name it `addRole`

File: libs/contract/src/lib/version/form/price/price.component.html [hunk 4]
```diff
@@ -0,0 +1,9 @@
+
+<mat-form-field>
+  <mat-label>Currency</mat-label>
+  <mat-select >
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/version/form/price/price.component.html]: Bind the select to a form

File: libs/contract/src/lib/version/form/price/price.component.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PriceComponent implements OnInit {
+  @Input() form: ContractVersionPriceForm;
+
+  public staticCurrencies = staticModels.MOVIE_CURRENCIES;
+
+  constructor() { }
+
+  ngOnInit() {
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/version/form/price/price.component.ts]: Remove if unused

File: libs/contract/src/lib/version/form/price/price.form.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { FormStaticValue } from '@blockframes/utils/form';
+import { FormControl } from '@angular/forms';
+import { Price } from '@blockframes/utils/common-interfaces/price';
+import { FormEntity } from '@blockframes/utils';
+
+function createContractVersionPriceControls(price: Partial<Price>){
+    return {
+        amount: new FormControl(price.amount),
+        currency: new FormStaticValue(price.currency, 'MOVIE_CURRENCIES')
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/version/form/price/price.form.ts]: 👍

File: libs/contract/src/lib/contract/form/party-name/party-name.component.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+export class ContractFormPartyNameComponent {
+  @Input() form: FormList<any, PartyDetailsForm>;
+  @Input() type: 'licensee' | 'licensor' = 'licensee';
+  
+  public addRole() {
+    this.form.add({ party: { role: this.type } });
+  }
+
+  public showTooltip(type: 'add' | 'remove') {
+    const titlecase = this.type.charAt(0).toUpperCase() + this.type.slice(1)
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/contract/form/party-name/party-name.component.ts]: Looks totally overkill ^^. But ok

Comment (Maxstgt) [reply libs/contract/src/lib/contract/form/party-name/party-name.component.ts]: if you got a slimer way of uppercase the first char, my ears are yours

Comment (GrandSchtroumpf) [reply libs/contract/src/lib/contract/form/party-name/party-name.component.ts]: In a normal situation, I would use css: text-transform
But here I would just right the code as a static string in the html.

**Pull request merged**